docs/architecture/providers/parameterized.md

```diff
@@ -72,3 +86,19 @@ for example).
 
 (replacement-extension-providers)=
 ## Replacement and extension parameterization
+
+Currently, the only kind of parameterization is *replacement parameterization*.
+In this mode, once the provider has been parameterized, it will only be used for
+that parameter only; and not without the parameter or with another parameter.
```

Comment:
Thanks for adding this. Such a simple thing we should have written down in the first place!